src/ucp/amo/amo.inl

```diff
@@ -178,62 +216,65 @@ ucp_amo_send_request(ucp_request_t *req, ucp_send_callback_t cb)
 }
 
 static inline
-uct_pending_callback_t ucp_amo_select_uct_func(ucp_atomic_fetch_op_t opcode, size_t op_size)
+uct_pending_callback_t ucp_amo_fetch_select_uct_func(ucp_atomic_fetch_op_t opcode, size_t op_size)
 {
-    uct_pending_callback_t progress_func;
-
     if (op_size == sizeof(uint64_t)) {
         switch (opcode) {
         case UCP_ATOMIC_FETCH_OP_CSWAP:
-            progress_func = _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_cswap64);
-            break;
-        case UCP_ATOMIC_FETCH_OP_SWAP:
-            progress_func = _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_swap64);
-            break;
+            return _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_cswap64);
         case UCP_ATOMIC_FETCH_OP_FADD:
-            progress_func = _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_fadd64);
-            break;
+        case UCP_ATOMIC_FETCH_OP_FAND:
+        case UCP_ATOMIC_FETCH_OP_FOR:
+        case UCP_ATOMIC_FETCH_OP_FXOR:
+        case UCP_ATOMIC_FETCH_OP_SWAP:
+            return _UCP_PROGRESS_AMO_NAME(uct_ep_atomic64_fetch);
         default:
-            progress_func = NULL;
+            return NULL;
         }
     } else {
+        ucs_assert(op_size == sizeof(uint32_t));
+
         switch (opcode) {
         case UCP_ATOMIC_FETCH_OP_CSWAP:
-            progress_func = _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_cswap32);
-            break;
-        case UCP_ATOMIC_FETCH_OP_SWAP:
-            progress_func = _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_swap32);
-            break;
+            return _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_cswap32);
         case UCP_ATOMIC_FETCH_OP_FADD:
-            progress_func = _UCP_PROGRESS_AMO_NAME(uct_ep_atomic_fadd32);
-            break;
+        case UCP_ATOMIC_FETCH_OP_FAND:
+        case UCP_ATOMIC_FETCH_OP_FOR:
+        case UCP_ATOMIC_FETCH_OP_FXOR:
+        case UCP_ATOMIC_FETCH_OP_SWAP:
+            return _UCP_PROGRESS_AMO_NAME(uct_ep_atomic32_fetch);
         default:
-            progress_func = NULL;
+            return NULL;
         }
     }
-    return progress_func;
 }
 
 static inline
 uct_pending_callback_t ucp_amo_post_select_uct_func(ucp_atomic_post_op_t opcode, size_t op_size)
 {
-    uct_pending_callback_t progress_func;
+    if (op_size == sizeof(uint64_t)) {
+        switch (opcode) {
```

Comment:
why need switch/case here? UCP_UCT_AMO_OPCODE_POST already checks for errors..

Reply:
fixed